src/components/ResourceBrowser/ResourceList/Sidebar.tsx

```diff
@@ -272,14 +187,32 @@ const Sidebar = ({
     }
 
     const handleInputChange = (newValue: string, actionMeta: InputActionMeta): void => {
-        if (actionMeta.action === 'input-change') {
-            setSearchText(newValue)
-            setMenuOpen(!!newValue)
+        // TODO: replace with enum after merge of feat/user-status-p3
+        if (actionMeta.action !== 'input-change') {
+            return
+        }
+
+        setSearchText(newValue)
+    }
+
+    const bringMatchedAbbreviatedOptionToFront = (options: K8sObjectOptionType[]): K8sObjectOptionType[] => {
+        if (!searchText || !k8Abbreviates) {
```

Comment:
Can we clean the conditions by moving searchText?.toLowerCase() above and encapsulate final conditions into better names